distributed/tests/test_diskutils.py

```diff
@@ -286,3 +287,22 @@ def test_workspace_concurrency(tmpdir):
 
     # We attempted to purge most directories at some point
     assert n_purged >= 0.5 * n_created > 0
+
+
+@pytest.mark.skipif(WINDOWS, reason="Need POSIX filesystem permissions and UIDs")
+def test_unwritable_base_dir():
+    with tempfile.TemporaryDirectory() as tmpdir:
```

Comment:
I tried using the tmpdir pytest fixture, but it was complaining about unwriteable directories on cleanup

Reply:
I'm not sure that specific error, but FWIW the [`dask.utils.tmpdir`](https://github.com/dask/dask/blob/72832d4bd685a25000ee646966a1aa5d0d8a1793/dask/utils.py#L244-L245) context manager seems to work well on Windows

Reply:
I can't provoke any complaints (on either macos or linux) with:

```python
import pytest
import os


def test_foo(tmpdir):
    os.mkdir(tmpdir / "bad", mode=0o500)

    with pytest.raises(PermissionError):
        open(tmpdir / "bad" / "tryme", "w")
```

Reply:
Weird, I can no longer reproduce the error. I've changed to tmpdir.